src/mcts/search.cc

```diff
@@ -603,6 +589,19 @@ void SearchWorker::ExtendNode(Node* node) {
     }
   }
 
+  auto legal_moves = board.GenerateLegalMoves();
```

Comment:
If no-capture-ply is 100 and there is a checkmate on a board, it's still checkmate, so that won't work.